src/test/java/net/snowflake/ingest/streaming/internal/StreamingIngestBigFilesIT.java

```diff
@@ -0,0 +1,199 @@
+
+    prop = TestUtils.getProperties(bdecVersion);
+    if (prop.getProperty(ROLE).equals("DEFAULT_ROLE")) {
+      prop.setProperty(ROLE, "ACCOUNTADMIN");
+    }
+    client =
+        (SnowflakeStreamingIngestClientInternal<?>)
+            SnowflakeStreamingIngestClientFactory.builder("client1").setProperties(prop).build();
+  }
+
+  @After
+  public void afterAll() throws Exception {
+    client.close();
+    jdbcConnection.createStatement().execute(String.format("drop database %s", testDb));
+  }
+
+  @Test
+  public void testManyRowsMultipleChannelsToMultipleTable()
+      throws SQLException, ExecutionException, InterruptedException {
+    String tableNamePrefix = "t_big_table_";
+
+    int numTables = 2;
+    int numChannels = 20; // channels are assigned round-robin to tables.
+    int batchSize = 10000;
+    int numBatches = 10; // number of rows PER CHANNEL is batchSize * numBatches
+    boolean isNullable = false;
+
+    Map<Integer, Integer> tableIdToNumChannels = new HashMap<>();
+    for (int i = 0; i < numChannels; i++) {
+      tableIdToNumChannels.put(
+          i % numTables, tableIdToNumChannels.getOrDefault(i % numTables, 0) + 1);
+    }
+    for (int i = 0; i < numTables; i++) {
+      String tableName = tableNamePrefix + i;
+      createTableForTest(tableName);
+    }
+
+    ingestRandomRowsToTable(
+        tableNamePrefix, numTables, numChannels, batchSize, numBatches, isNullable);
+
+    for (int i = 0; i < numTables; i++) {
+      int numChannelsToTable = tableIdToNumChannels.get(i);
+      verifyTableRowCount(
+          batchSize * numBatches * numChannelsToTable,
+          jdbcConnection,
+          testDb,
+          TEST_SCHEMA,
+          tableNamePrefix + i);
+
+      // select * to ensure scanning works
+      ResultSet result =
+          jdbcConnection
+              .createStatement()
+              .executeQuery(
+                  String.format(
+                      "select * from %s.%s.%s", testDb, TEST_SCHEMA, tableNamePrefix + i));
+      result.next();
+      Assert.assertNotNull(result.getString("STR"));
+    }
+  }
+
+  private void ingestRandomRowsToTable(
+      String tablePrefix,
+      int numTables,
+      int numChannels,
+      int batchSize,
+      int iterations,
+      boolean isNullable)
+      throws ExecutionException, InterruptedException {
+
+    List<Map<String, Object>> rows = new ArrayList<>();
+    for (int i = 0; i < batchSize; i++) {
+      Random r = new Random();
+      rows.add(TestUtils.getRandomRow(r, isNullable));
+    }
+
+    ExecutorService testThreadPool = Executors.newFixedThreadPool(numChannels);
+    CompletableFuture[] futures = new CompletableFuture[numChannels];
+    List<SnowflakeStreamingIngestChannel> channelList = new ArrayList<>();
+    for (int i = 0; i < numChannels; i++) {
+      final String channelName = "CHANNEL" + i;
+      int finalI = i;
+      futures[i] =
+          CompletableFuture.runAsync(
```

Comment:
just curious does it have to be concurrent ingest to produce big files?

Reply:
yes, I tried the same configuration but not concurrent and it produced many small BDEC files, < 20 MB.